tests/unit/models/NotificationConfiguration.js

```diff
@@ -0,0 +1,216 @@
+const assert = require('assert');
+const { parseString } = require('xml2js');
+
+const NotificationConfiguration =
+    require('../../../lib/models/NotificationConfiguration.js');
+
+function checkError(parsedXml, err, errMessage, cb) {
+    const config = new NotificationConfiguration(parsedXml).
+        getValidatedNotificationConfiguration();
+    assert.strictEqual(config.error[err], true);
+    assert.strictEqual(config.error.description, errMessage);
+    cb();
+}
+
+function generateEvent(testParams) {
+    const event = [];
+    if (testParams.key === 'Event') {
+        if (Array.isArray(testParams.value)) {
+            testParams.value.forEach(v => {
+                event.push(`${event}<Event>${v}</Event>`);
+            });
+        } else {
+            event.push(`<Event>${testParams.value}</Event>`);
+        }
+    } else {
+        event.push('<Event>s3:ObjectCreated:*</Event>');
+    }
+    return event.join('');
+}
+
+function generateFilter(testParams) {
+    let filter = '';
+    if (testParams.key === 'Filter') {
```

Comment:
I would move these into functions where `.key` is the key on an object and the value is the function. Something like:

```javascript
const filterMapper = {
    "Filter": test => `<Filter>...`,
    "S3Key: test => `you get the idea`,
}

function generateFilter(testParams) {
    return filterMapper[testParams.key](testParams)
}
```